x-pack/plugins/security_solution/public/management/pages/trusted_apps/view/components/effected_policy_select/effected_policy_select.tsx

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { memo, useCallback, useEffect, useMemo, useState } from 'react';
+import {
+  EuiFormRow,
+  EuiSelectable,
+  EuiSelectableProps,
+  EuiSwitch,
+  EuiSwitchProps,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { EuiSelectableOption } from '@elastic/eui/src/components/selectable/selectable_option';
+import { PolicyData } from '../../../../../../../common/endpoint/types';
+
+interface OptionPolicyData {
+  policy: PolicyData;
+}
+
+type EffectedPolicyOption = EuiSelectableOption<OptionPolicyData>;
+
+export interface EffectedPolicySelection {
+  isGlobal: boolean;
+  selected: PolicyData[];
+}
+
+export type EffectedPolicySelectProps = Omit<
+  EuiSelectableProps<OptionPolicyData>,
+  'onChange' | 'options' | 'children' | 'searchable'
+> & {
+  options: PolicyData[];
+  isGlobal: boolean;
+  onChange: (selection: EffectedPolicySelection) => void;
+  selected?: PolicyData[];
+};
+export const EffectedPolicySelect = memo<EffectedPolicySelectProps>(
+  ({
+    isGlobal,
+    onChange,
+    listProps,
+    options,
+    selected = [],
+    'data-test-subj': dataTestSubj,
+    ...otherSelectableProps
+  }) => {
+    const DEFAULT_LIST_PROPS = useMemo(() => ({ bordered: true }), []);
+
+    const [selectionState, setSelectionState] = useState<EffectedPolicySelection>({
+      isGlobal,
+      selected,
+    });
+
+    const getTestId = useCallback(
+      (suffix): string | undefined => {
+        if (dataTestSubj) {
+          return `${dataTestSubj}-${suffix}`;
+        }
+      },
+      [dataTestSubj]
+    );
```

Comment:
I saw you are doing this often, maybe deserves a custom hook? that would even take care of extracting 'data-test-subj' from props?

Reply:
Yeah, It does seem like i have done it a few times now - I think mainly because we been createing these compound components. I will look into splitting it off into a hook

Reply:
Snapshot testing avoids the need for this :P